demisto_sdk/commands/test_content/TestContentClasses.py

```diff
@@ -1872,18 +1872,24 @@ def _poll_for_playbook_state(self) -> str:
         """
         timeout = time.time() + self.playbook.configuration.timeout
         number_of_attempts = 1
+
+        should_re_authenticate = True
         # wait for playbook to finish run
         while True:
             # give playbook time to run
             time.sleep(5)
             try:
                 # fetch status
                 playbook_state = self._get_investigation_playbook_state()
-            except demisto_client.demisto_api.rest.ApiException:
-                playbook_state = "Pending"
-                self.build_context.logging_module.exception(
-                    "Error when trying to get investigation playbook state"
-                )
+            except demisto_client.demisto_api.rest.ApiException as e:
+                if e.status == 401 and should_re_authenticate:
+                    self.server_context._configure_new_client()
+                    should_re_authenticate = False
+                else:
+                    playbook_state = "Pending"
+                    self.build_context.logging_module.exception(
+                        "Error when trying to get investigation playbook state"
+                    )
```

Comment:
```suggestion
            except demisto_client.demisto_api.rest.ApiException as e:
                if e.status == 401 and should_re_authenticate:
                    self.server_context._configure_new_client()
                    self.build_context.logging_module.debug('new demisto_client created')
                    continue
                playbook_state = "Pending"
                self.build_context.logging_module.exception(
                    "Error when trying to get investigation playbook state"
                )
```

Reply:
@esharf do you want to each time we receive 401, we will reset the client? not only the first time it occurs?  

this should be:

```
if e.status == 401:
```

cause `should_re_authenticate` is removed